pkg/pillar/cmd/zedrouter/probe.go

```diff
@@ -287,25 +291,15 @@ func portGetIntfAddr(port types.NetworkPortStatus) net.IP {
 // local and remote(less frequent) host probing
 func launchHostProbe(ctx *zedrouterContext) {
 	var isReachable, needSendSignal, bringIntfDown bool
-	var remoteURL string
 	nhPing := make(map[string]bool)
 	localDown := make(map[string]bool)
 	remoteProbe := make(map[string]map[string]probeRes)
 	log.Infof("launchHostProbe: enter\n")
 	dpub := ctx.subDeviceNetworkStatus
 	ditems := dpub.GetAll()
 
-	if serverNameAndPort == "" {
-		server, err := ioutil.ReadFile(types.ServerFileName)
-		if err == nil {
-			serverNameAndPort = strings.TrimSpace(string(server))
-		}
-	}
-	if serverNameAndPort != "" {
-		remoteURL = serverNameAndPort
-	} else {
-		remoteURL = "www.google.com"
-	}
+	probeMutex.Lock()
```

Comment:
Why are you holding the mutex while opening and reading a file?

Reply:
will move this Lock after the getSystemURL() call.